src/gtc/passes/oir_access_kinds.py

```diff
@@ -37,7 +37,8 @@ def visit_ScalarAccess(self, node: oir.ScalarAccess, **kwargs: Any) -> None:
         self._visit_Access(node.name, **kwargs)
 
     def visit_FieldAccess(self, node: oir.FieldAccess, **kwargs: Any) -> None:
-        self.generic_visit(node, **kwargs)
+        self.visit(node.offset, **{**kwargs, "kind": AccessKind.READ})
+        self.visit(node.data_index, **{**kwargs, "kind": AccessKind.READ})
```

Comment:
Is it possible/expected that `kwargs` has a `kind` item? If it is not possible, I would write this in a simpler way:
```suggestion
        self.visit(node.offset, kind=AccessKind.READ, **kwargs)
        self.visit(node.data_index, kind=AccessKind.READ, **kwargs)
```
Additionally, why have been these two functions added here?

Reply:
`kind` is sometimes in the dict when it comes directly from the `visit_AssignStmt`.